### PR TITLE
Allow usage of the "adjust forecast" feature

### PR DESCRIPTION
**Pull request by GreenOpposite**
This adds the "actual" query parameter to the estimate method, needed to use forecast.solars "adjust forecast" feature.
This as a first towards implementing [this feature request](https://community.home-assistant.io/t/forecast-solar-optional-send-data-in-get-requests/629216).

### CODE REVIEW

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]
`actual` can be added to the params and does not need to be added to the end of the url.

**Comment by ximex** [comment]
any updates to this PR when this PR is planed to get merged/released?

**Comment by klaasnicolaas** [comment]
When all the CI checks are passing 😉 It needs to add tests for full pytest coverage

./Klaas

**Comment by ximex** [comment]
@GreenOpposite could you check that? I'm not a python dev

**Comment by GreenOpposite** [comment]
When I run the tests locally, everything passes and I get a code coverage of 95%, but the automatic test on GitHub fail to run with the error `Error: Codecov token not found. Please provide Codecov token with -t flag.`

**Comment by GreenOpposite** [comment]
After a bit of investigation, it seems like what's happening is that the Codecov GitHub integration needs a token that my forked repository is missing. A maintainer has to approve the workflows for them to run in the context of this repository. @klaasnicolaas

**Comment by GreenOpposite** [comment]
Any updates on the state of this PR? Do I need to do something? @klaasnicolaas

**Comment by klaasnicolaas** [comment]
What I'm wondering is whether we should pass the `actual` parameter with every request by default (if you have an API key), or only when the value is higher than 0. 

The only thing I'm missing is a test that tests what happens if you specify a value higher than 0.

./klaas

**Comment by GreenOpposite** [comment]
A value of `0` is also used to signal invalidation/reset of the previously send correction data. So not passing an `actual` value to the `estimate()` function has the effect of deactivating the correction.
```python
a = estimate()
b = estimate(actual=15)
c = estimate()
```
If you don't send `0` every time, then `a` and `c` would have different values, which I would argue is not intuitive.

What kind of testing is missing exactly?

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]
A test where the `actual` is given a value higher than 0 and has an apikey. It may not be needed for the codecov, but then it is fully tested.

**Comment by klaasnicolaas** [comment]
Another thing I was wondering: suppose someone has solar panels in 2 zones and therefore also a split forecast.solar, to what extent will this have an effect if you send the full actual of both zones together (because you often have that as an entity) every time. 🤔 

./Klaas

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]
I think we should send float values ​​and these values ​​are a bit more realistic

File: tests/test_models.py [hunk 1]
```diff
@@ -103,6 +103,41 @@ async def test_estimated_forecast_with_subscription(
...
+            status=200,
+            headers={
+                "Content-Type": "application/json",
+                "X-Ratelimit-Limit": "60",
+                "X-Ratelimit-Period": "3600",
+            },
+            text=load_fixtures("forecast_personal.json"),
+        ),
+    )
+    forecast: Estimate = await forecast_key_client.estimate(actual=5800)
```
[the remark this answers is not in the document]
Comment (GreenOpposite) [reply tests/test_models.py:127]: True 🙃. Those values make a lot more sense!

**Comment by GreenOpposite** [comment]
> Another thing I was wondering: suppose someone has solar panels in 2 zones and therefore also a split forecast.solar, to what extent will this have an effect if you send the full actual of both zones together (because you often have that data as single entity) every time. 🤔
> 
> ./Klaas

I think that might lead to wrong values being predicted, but I also don't see how that could be easily mitigated. You would need separate data for each zone and send the appropriate data with each request.

**Review by klaasnicolaas — APPROVED** [review]

**Comment by dennisreimann** [comment]
Will this be supported/used in the HA integration?

**Pull request merged**